index.js

```diff
@@ -0,0 +1,91 @@
+let slideIndex = 0;
+
+function moveSlide(n) {
+  const slides = document.querySelectorAll('.carousel__item');
```

Comment:
Para interactuar con elementos de la pagina usa clases que comiencen con `.js`
```suggestion
  const slides = document.querySelectorAll('.js-carousel-item');
```
El elemento lo escribes entonces asi:
`<div class="carousel__item product-banner js-carousel-item" style="background-image: url('image/Banner.jpg')"></div>`